firebase/functions/src/index.ts

```diff
@@ -0,0 +1,29 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+
+admin.initializeApp();
+
+const db = admin.firestore();
+
+exports.modifyRating = functions.firestore
+    .document('ta-course/{taCourseId}/ratings/{uid}')
+    .onWrite(async (change, context) => {
+
+        const taCourseId = context.params.taCourseId;
+        const uid = context.params.uid;
+
+        const newData = change.after.exists ? change.after.data() : null;
+        const oldData = change.before.exists ? change.before.data() : null;
+
+        const taCourseDoc = db.collection('ta-course').doc(taCourseId)
+
+        let newTaCourse: any = {};
+
+        if (newData && newData.rating)
+            newTaCourse['rating' + newData.rating.toString()] = admin.firestore.FieldValue.arrayUnion(uid);
```

Comment:
why and object prop could have value 'rating5'?) Or could you explain how does this work?

Reply:
I think it is better to show. This is how the doc looks like. Each rating (1-5) is a list of user ids of users who provided the rating.
![image](https://user-images.githubusercontent.com/8573325/101667513-79394580-3a60-11eb-9c1c-a249ae3b8b7c.png)